### PR TITLE
Add Activities page and model

### PR DESCRIPTION
**Pull request by comzeradd**
That implements the Activities page based on the #287 design.

The only thing currently missing is a dynamic image per Activity.

### CODE REVIEW

File: coss/opensource_clubs/templates/blocks/activity_card.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{% load static wagtailcore_tags %}
+
+<div class="card card-mini">
+  <h4 class="card-header">{{ value.title }}</h4>
+  <div class="card-body">
+    {{ value.text|richtext }}
```
Comment (akatsoulas) [inline coss/opensource_clubs/templates/blocks/activity_card.html]: I am not sure if `|richtext` is needed here.

File: coss/opensource_clubs/models.py [hunk 2]
```diff
@@ -129,3 +129,16 @@ class FAQPage(Page):
     content_panels = Page.content_panels + [
         StreamFieldPanel('faq'),
     ]
+
+
+class ActivitiesPage(Page):
+    """Activities Page for Open Source Clubs."""
+    description = RichTextField(blank=True)
```
Comment (akatsoulas) [inline coss/opensource_clubs/models.py]: We can make the whole card a streamfield which we can also use for the collaborators section.

Comment (akatsoulas) [reply coss/opensource_clubs/models.py]: Regardless of the streamfield, the design has a photo section too. Let's add this functionality and if no photo is provided we can default to the one listed in the design.

File: coss/opensource_clubs/templates/opensource_clubs/faq_page.html [hunk 3]
```diff
@@ -10,13 +10,16 @@
...
 {% endblock %}
 
+{% block jumbotron %}
+{% endblock %}
+
 {% block extra_about %}
   <div class="faq">
     <div class="container">
-      <h4>FAQ</h4>
+      <h3>{{ page.title }}</h3>
```
Comment (akatsoulas) [inline coss/opensource_clubs/templates/opensource_clubs/faq_page.html]: Good catch.

**Comment by comzeradd** [comment]
PR updated. I added an optional image field to the `InputTextContentBlock` content block.

**Comment by akatsoulas** [comment]
I would prefer a more DRY approach. Adding an Image in an InputText streamfield probably it's not the best way forward. You can either create a new streamfield for the activity, reusable to other places too (preferred option for me) or add the Image as a foreign key to the activity page.

**Comment by comzeradd** [comment]
PR updated. I created a new content block that extends `InputTextContentBlock` to add an image field. We easily used that for other card-based content, like collaborators.

File: coss/opensource_clubs/models.py [hunk 4]
```diff
@@ -129,3 +129,16 @@ class FAQPage(Page):
     content_panels = Page.content_panels + [
         StreamFieldPanel('faq'),
     ]
+
+
+class ActivitiesPage(Page):
+    """Activities Page for Open Source Clubs."""
```
Comment (akatsoulas) [inline coss/opensource_clubs/models.py]: Doesn't this need a title?

File: coss/opensource_clubs/models.py [hunk 5]
```diff
@@ -129,3 +129,16 @@ class FAQPage(Page):
...
+    description = RichTextField(blank=True)
+    activity = StreamField([
+        ('info', CardContentBlock(required=True, template='blocks/activity_card.html'),)
```
Comment (akatsoulas) [inline coss/opensource_clubs/models.py]: Why don't you add the `template` in the `Meta` class of the stremfield?

File: coss/opensource_clubs/templates/blocks/activity_card.html [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+<div class="card card-mini">
+  {% if value.image %}
+    {% image value.image fill-180x180 class="rounded-circle card-img" %}
+  {% else %}
+    <img src="{% static 'img/clubs-illustration.svg'%}" class="rounded-circle card-img" alt="activity">
+  {% endif %}
+  <h4 class="card-header">{{ value.title }}</h4>
+  <div class="card-body">
+    {{ value.text|richtext }}
```
Comment (akatsoulas) [inline coss/opensource_clubs/templates/blocks/activity_card.html]: Is there a specific reason on why don't you use the text input block here and you are re-adding these entries?

File: coss/opensource_clubs/templates/opensource_clubs/activities_page.html [hunk 7]
```diff
@@ -0,0 +1,31 @@
+{% extends "home/about_page.html" %}
```
Comment (akatsoulas) [inline coss/opensource_clubs/templates/opensource_clubs/activities_page.html]: Why is this needed?

File: coss/opensource_clubs/models.py (hunk 4 above)
Comment (comzeradd) [reply coss/opensource_clubs/models.py]: Nope. Since it inherits from `Page` it already has a title field.

File: coss/opensource_clubs/models.py (hunk 5 above)
Comment (comzeradd) [reply coss/opensource_clubs/models.py]: We could do that, but I thought that this card block can also be used for other cases where we may have slightly different UI. So it's probably better to maintain that flexibility.

Comment (akatsoulas) [reply coss/opensource_clubs/models.py]: My guess is that even if you have a template `foo` in the meta class, adding a template `bar` here would override it. Is this true? 
Moreover, having a default template in the Meta class is easier and less prone to errors than having to remember every time to declare one. If you have specific needs that the default doesn't cover then you will definitely remember to override it.

File: coss/opensource_clubs/templates/opensource_clubs/activities_page.html (hunk 7 above)
Comment (comzeradd) [reply coss/opensource_clubs/templates/opensource_clubs/activities_page.html]: Same as the FAQ page, this inherits from the about_page template. We should probably refactor this to have a more generic name, since we use it on various other occasions besides the About.

**Comment by comzeradd** [comment]
PR updated. I used your suggestion and moved the block template under the home app.
